recipes/msix/all/conanfile.py

```diff
@@ -0,0 +1,130 @@
+import os
+from conans import CMake, ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+
+
+class MsixConan(ConanFile):
+    name = "msix"
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/microsoft/msix-packaging"
+    description = "An SDK for creating MSIX packages"
+    topics = ("msix", "sdk", "packaging", "conan-recipe")
+
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "crypto_lib": ["crypt32", "openssl"],
+        "pack": [True, False],
+        "skip_bundles": [True, False],
+        "use_external_zlib": [True, False],
+        "use_validation_parser": [True, False],
+        "xml_parser": ["applexml", "javaxml", "msxml6", "xerces"]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "crypto_lib": "openssl",
+        "pack": False,
+        "skip_bundles": False,
+        "use_external_zlib": True,
+        "use_validation_parser": False,
+        "xml_parser": "xerces"
+    }
+
+    generators = "cmake"
+    exports_sources = "CMakeLists.txt", "patches/**"
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        if self.settings.os == "Android":
+            self._cmake.definitions["AOSP"] = True
+        if self.settings.os == "Linux":
+            self._cmake.definitions["LINUX"] = True
+        if self.settings.os == "Macos":
+            self._cmake.definitions["MACOS"] = True
+        self._cmake.definitions["CRYPTO_LIB"] = self.options.crypto_lib
+        self._cmake.definitions["MSIX_PACK"] = self.options.pack
+        self._cmake.definitions["MSIX_SAMPLES"] = False
+        self._cmake.definitions["MSIX_TESTS"] = False
+        self._cmake.definitions["SKIP_BUNDLES"] = self.options.skip_bundles
+        self._cmake.definitions["USE_EXTERNAL_ZLIB"] = self.options.use_external_zlib
+        self._cmake.definitions["USE_SHARED_ZLIB"] = self.options["zlib"].shared
+        self._cmake.definitions["USE_VALIDATION_PARSER"] = self.options.use_validation_parser
+        self._cmake.definitions["XML_PARSER"] = self.options.xml_parser
+        self._cmake.configure()
+        return self._cmake
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.settings.os == "Linux" and (self.settings.compiler != "clang"):
+            raise ConanInvalidConfiguration("Only clang is supported on Linux")
```

Comment:
Move all of those invalid configuration checks to `validate()`

https://docs.conan.io/en/latest/reference/conanfile/methods.html#validate

Reply:
[Done](https://github.com/conan-io/conan-center-index/pull/6475/commits/9720eefcded0821865cca5d8880c0c30da06aaa4)